lib/quicrq_internal.h

```diff
@@ -371,6 +376,64 @@ typedef struct st_quicrq_notify_url_t {
     uint8_t* url;
 } quicrq_notify_url_t;
 
+/* Context representing unidirectional streams*/
+struct st_quicrq_uni_stream_ctx_t {
+    struct st_quicrq_uni_stream_ctx_t* next_stream;
+    struct st_quicrq_uni_stream_ctx_t* previous_stream;
+    struct st_quicrq_cnx_ctx_t* cnx_ctx;
+    /* Source from which data is read and sent on the stream. */
+    quicrq_media_source_ctx_t* media_source;
+    struct st_quicrq_uni_stream_ctx_t* next_stream_for_source;
+    struct st_quicrq_uni_stream_ctx_t* previous_stream_for_source;
+
+    /* Control stream context */
+    struct st_quicrq_stream_ctx_t* control_stream;
+
+    uint64_t stream_id;
+    /* media_id: local identifier of media stream.*/
+    uint64_t media_id;
+    uint64_t next_group_id;
+    uint64_t next_object_id;
+    uint64_t next_object_offset;
+    uint64_t current_group_id;
+    uint64_t start_object_id;
+
+    /* Stream state */
+    quicrq_stream_sending_state_enum send_state;
+    quicrq_stream_receive_state_enum receive_state;
+    /* Close reason and diagnostic code */
+    quicrq_media_close_reason_enum close_reason;
+    uint64_t close_error_code;
+    /* Control flags */
+    unsigned int is_sender : 1;
+    /* is_cache_real_time:
+     * Indicates whether local cache management follows the "real time" logic,
+     * in which only recent objects are kept. By default, cache management
+     * follows the "streaming" logic, in which everything is kept -- or nothing.
+     */
+    unsigned int is_cache_real_time : 1;
+    /* is_peer_finished, is_local_finished, is_receive_complete:
+     * For the sender, receiver finished happens if the client closes the control stream.
+     * In that case, the server should close the stream and mark itself finished.
+     * For the receiver, the transfer finishes if everything was received. In that
+     * case, the receiver shall close the control stream. If the sender closes the
+     * control stream before that, we have an abnormal close.
+     */
+    unsigned int is_peer_finished : 1;
+    unsigned int is_local_finished : 1;
+    unsigned int is_receive_complete: 1;
+    unsigned int is_active_datagram : 1;
+    unsigned int is_start_object_id_sent : 1;
+    unsigned int is_final_object_id_sent : 1;
+    unsigned int is_cache_policy_sent : 1;
+
+    quicrq_message_buffer_t message_sent;
+    quicrq_message_buffer_t message_receive;
+
+    quicrq_media_consumer_fn consumer_fn; /* Callback function for media data arrival  */
+};
```

Comment:
Do not replicate data from the control stream, because they could become inconsistent! For the call, you should just use: `uni_stream_ctx->control_stream->consumer_fn`.